scripts/system/create/entitySelectionTool/entitySelectionTool.js

```diff
@@ -670,7 +671,104 @@ SelectionManager = (function() {
             that._update(false, this);
         } else {
             audioFeedback.rejection();
+            Window.notifyEditError("You have nothing selected or the selection has locked entities.");
+        }
+    };
+
+    that.selectParent = function() {
+        if (that.hasSelection()) {
+            var currentSelection = that.selections;
+            that.selections = [];
+            for (var i = 0; i < currentSelection.length; i++) {
+                var properties = Entities.getEntityProperties(currentSelection[i], ['parentID']);
+                if (properties.parentID !== Uuid.NULL) {
+                    that.selections.push(properties.parentID);
+                }
+            }
+            that._update(true, this);
+        } else {
+            audioFeedback.rejection();
+            Window.notifyEditError("You have nothing selected.");            
+        }
+    };
+
+    that.selectTopParent = function() {
+        if (that.hasSelection()) {
+            var currentSelection = that.selections;
+            that.selections = [];
+            for (var i = 0; i < currentSelection.length; i++) {
+                var topParentId = getTopParent(currentSelection[i]);
+                if (topParentId !== Uuid.NULL) {
+                    that.selections.push(topParentId);
+                }
+            }
+            that._update(true, this);
+        } else {
+            audioFeedback.rejection();
+            Window.notifyEditError("You have nothing selected.");            
+        }
+    };
+
+    function getTopParent(id) {
+        var topParentId = Uuid.NULL;
+        var properties = Entities.getEntityProperties(id, ['parentID']);
+        if(properties.parentID === Uuid.NULL) {
+            topParentId = id;
+        } else {
+            topParentId = getTopParent(properties.parentID);
         }
+        return topParentId;
+    }
+
+    that.addChildrenToSelection = function() {
+        if (that.hasSelection()) {
+            for (var i = 0; i < that.selections.length; i++) {
+                var childrenIds = Entities.getChildrenIDs(that.selections[i]);
```

Comment:
```suggestion
                var childrenIDs = Entities.getChildrenIDs(that.selections[i]);
```
`childrenIDs` would be a nicer name, I think.